### PR TITLE
feat(support-matrix): add nodeOmit to drop EOL node majors from the card

### PR DESCRIPTION
**Pull request by Brooooooklyn**
## What

Adds a `nodeOmit` query param to the support-matrix badge service: a comma list of Node majors to drop from the Node.js card's pills, even when the `engines` range still permits them (e.g. an EOL runtime like Node 25 under `>=25`).

- `nodeOmit=25` turns pills `22.20+ · 24.12+ · 25 · 26` into `22.20+ · 24.12+ · 26`.
- The `engines` string is still rendered **verbatim**, so the declared range is never hidden.
- Sanitized exactly like `nodeTested` (whole-digit integers only; junk dropped).

## Consistency

Omission is applied **before** coverage/gap accounting, and the headline now spans the surviving pills, so:

- an omitted major leaves **no pill and no `excluded` mention** — it vanishes from the card;
- omitting the floor or ceiling major no longer dangles the headline (previously `>=22` + `nodeOmit=26` still showed `v22 → v26` with no `26` pill; now it retracts to `v22 → v25`).

## Threaded through

`lib/support-matrix/{node,query,resolve}.ts`, `routes.test.ts`, and the `/support-matrix` URL-builder (`components/support-matrix-builder/index.tsx`).

## Tests

`vitest` — node/query/resolve/routes suites, including boundary cases (omit floor, ceiling, all-covered majors, and an already-excluded major). Full suite: **367 passing**.

## Not included

Per-query **edge caching** for `/support-matrix.{png,svg}` is out of scope here — the void-platform direct-route cache key strips the query string, so it needs a platform-side fix (the static-candidate cache key), tracked separately.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/support-matrix/node.ts [hunk 1]
```diff
@@ -183,11 +188,14 @@ export function deriveNode(
...
     floor.minor > 0 ? `v${floor.major}.${floor.minor}` : `v${floor.major}`
-  const headline = floor.major < ceiling ? `${left} → v${ceiling}` : left
 
   const pills: NodePill[] = []
   const gaps: string[] = []
   for (let major = floor.major; major <= ceiling; major++) {
+    // An explicitly omitted (e.g. EOL) major is dropped whole — before any
+    // coverage/gap accounting — so it leaves no pill AND no `excluded` mention,
+    // whether or not the range covered it. It simply vanishes from the card.
+    if (omit.has(major)) continue
```
[the remark this answers is not in the document]
Comment (Brooooooklyn) [reply lib/support-matrix/node.ts:206]: Fixed in 59f2a5f. Gaps are now buffered and only flushed once a *later* surviving pill bounds them on both sides; any uncovered run trailing past the final pill is discarded. So `^22 || ^26` with `nodeOmit=26` reports headline `v22` and `excluded: null` (23/24/25 sit beyond the surviving span), and `^22 || ^24 || ^26` omit 26 lists only `23`. Added two regression tests covering both.
